packages/aptos-wallet-adapter/src/WalletAdapters/FoxWallet.ts

```diff
@@ -0,0 +1,312 @@
+import { Types } from 'aptos';
+import {
+  WalletAccountChangeError,
+  WalletDisconnectionError,
+  WalletGetNetworkError,
+  WalletNetworkChangeError,
+  WalletNotConnectedError,
+  WalletNotReadyError,
+  WalletSignAndSubmitMessageError,
+  WalletSignMessageError,
+  WalletSignTransactionError
+} from '../WalletProviders/errors';
+import {
+  AccountKeys,
+  BaseWalletAdapter,
+  NetworkInfo,
+  scopePollingDetectionStrategy,
+  SignMessagePayload,
+  SignMessageResponse,
+  WalletAdapterNetwork,
+  WalletName,
+  WalletReadyState
+} from './BaseAdapter';
+
+interface IApotsErrorResult {
+  code: number;
+  name: string;
+  message: string;
+}
+
+type AddressInfo = { address: string; publicKey: string; authKey?: string };
+
+interface IFoxWallet {
+  connect: () => Promise<AddressInfo>;
+  account: () => Promise<AddressInfo>;
+  isConnected: () => Promise<boolean>;
+  signAndSubmitTransaction(
+    transaction: any,
+    options?: any
+  ): Promise<{ hash: Types.HexEncodedBytes } | IApotsErrorResult>;
+  signTransaction(transaction: any, options?: any): Promise<Uint8Array | IApotsErrorResult>;
+  signMessage(message: SignMessagePayload): Promise<SignMessageResponse>;
+  disconnect(): Promise<void>;
+  network(): Promise<WalletAdapterNetwork>;
+  requestId: Promise<number>;
+  onAccountChange: (listener: (newAddress: AddressInfo) => void) => void;
+  onNetworkChange: (listener: (network: { networkName: string }) => void) => void;
+}
+
+interface AptosWindow extends Window {
+  foxwallet?: IFoxWallet;
+  aptos?: IFoxWallet;
```

Comment:
i think window.aptos should be reserved for Petra wallet

Reply:
FoxWallet is mobile only, and our API is Petra compatible, so it's not wrong.  We will inject aptos under foxwallet in next release, then we'll change the adapter.